python/ray/test/test_ray_init.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import base64
+import os
+import pytest
+import redis
+
+import ray
+
+
+@pytest.fixture
+def password():
+    random_bytes = os.urandom(128)
+    if hasattr(random_bytes, "hex"):
+        return random_bytes.hex()  # Python 3
+    return random_bytes.encode("hex")  # Python 2
+
+
+@pytest.fixture
+def shutdown_only():
+    # Fix for https://github.com/ray-project/ray/issues/3045
+    global f
+    f = ray.remote(f._function)
+
+    yield None
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+@ray.remote
+def f():
+    return 1
+
+
+class TestRedisPassword(object):
+    @pytest.mark.skipif(
+        os.environ.get("RAY_USE_NEW_GCS") != "on"
+        and os.environ.get("RAY_USE_XRAY"),
+        reason="Redis authentication works for raylet and old GCS.")
+    def test_exceptions(self, password, shutdown_only):
+        with pytest.raises(Exception):
+            ray.init(redis_password=password)
+
+    @pytest.mark.skipif(
+        os.environ.get("RAY_USE_NEW_GCS") == "on",
+        reason="New GCS API doesn't support Redis authentication yet.")
+    @pytest.mark.skipif(
+        not os.environ.get("RAY_USE_XRAY"),
+        reason="Redis authentication is not supported in legacy Ray.")
+    def test_redis_password(self, password, shutdown_only):
+        info = ray.init(redis_password=password)
+        redis_address = info["redis_address"]
+        redis_ip, redis_port = redis_address.split(":")
+
+        # Check that we can run a task
+        task_id = f.remote()
+        ready, running = ray.wait([task_id], timeout=5000)
+        assert len(ready) > 0, "Expected task to complete"
```

Comment:
Kind of a nit, but `task_id` should be `object_id`.

Also, instead of doing `wait` followed by an assert, maybe just do `ray.get(object_id)`.

Reply:
Do you mind if I keep the `ray.wait`? While testing, I ran into a case where `ray.get` wouldn't fail, but block indefinitely (I think this was when I discovered the issues with the new GCS).